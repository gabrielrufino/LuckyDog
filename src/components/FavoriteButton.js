import Icon from 'react-native-vector-icons/FontAwesome5';
import React from 'react';
import {TouchableWithoutFeedback} from 'react-native';

export default function FavoriteButton({isFavorite, onPress}) {
  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <Icon
        name="star"
        size={30}
        regular={!isFavorite}
        solid={isFavorite}
        color="#EEC584"
      />
    </TouchableWithoutFeedback>
  );
}
