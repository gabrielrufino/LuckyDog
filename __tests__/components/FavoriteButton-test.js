import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import FavoriteButton from '../../src/components/FavoriteButton';

describe('components/FavoriteButton.js', () => {
  it('Should renders the favorite-button correctly', () => {
    const onPress = jest.fn();

    const tree = renderer.create(<FavoriteButton onPress={onPress} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
