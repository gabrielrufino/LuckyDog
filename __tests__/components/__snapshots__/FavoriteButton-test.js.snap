// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FavoriteButton.js Should renders the favorite-button correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "color": "#EEC584",
        "fontSize": 30,
      },
      undefined,
      Object {
        "fontFamily": "FontAwesome5Free-Regular",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {
        "fontWeight": "400",
      },
    ]
  }
>
  ï€…
</Text>
`;
